docs/source/examples/pyquil_demo.myst

```diff
@@ -0,0 +1,172 @@
+and one for the optimization routine to find the best parameters.
+ 
+```{code-cell}
+# define the expecation function used in the parameter scan
+def expectation_fn(theta, executable: Program) -> float:
+    bitstrings = qc.run(executable, memory_map={"theta": [theta]})
+    return np.mean(bitstrings[:, 0])
+```
++++
+
+```{code-cell}
+# define the expecation function used in the optimization routine
+def expectation(thetas, executable: Program) -> float:
+    bitstrings = qc.run(executable, memory_map={'theta': thetas})
+    result = np.mean(bitstrings[:, 0])
+    return -result
+```
++++
+
+Run the parameter scan without noise mitigation
+
+```{code-cell}
+program.wrap_in_numshots_loop(1000)
+quil_prog = qc.compiler.quil_to_native_quil(program, protoquil=True)
+
+thetas = np.linspace(0, 2 * np.pi, 21)
+results = []
+for theta in thetas:
+    results.append(expectation_fn(theta, quil_prog))
+```
++++
+
+Plot the energy landscape without noise mitigation
+```{code-cell}
+from matplotlib import pyplot as plt
+plt.figure(1)
+plt.plot(thetas, results, 'o-')
+plt.xlabel(r'$\theta$', fontsize=18)
+_ = plt.ylabel(r'$\langle \Psi(\theta) | \frac{1 - Z}{2} | \Psi(\theta) \rangle$', fontsize=18)    
+plt.title('Noisy Energy Landscape')
+```
++++
+
+Run optimization without noise mitigation
+
+```{code-cell}
+quil_prog = qc.compiler.quil_to_native_quil(program, protoquil=True)
+init_angle = [0.1]
+res = optimize.minimize(expectation, init_angle, args=(quil_prog), method='COBYLA')
+print(res)
+```
++++
+
+Now run parameter scan with ZNE
+
+```{code-cell}
+quil_prog = qc.compiler.quil_to_native_quil(program, protoquil=True)
+init_angle = [0.1]
+results_zne = []
+# here we use a linear fit for the zero noise extrapolation
+fac = mitiq.zne.inference.LinearFactory(scale_factors=[1.0, 3.0])
+for theta in thetas:
+    results_zne.append(
+        zne.execute_with_zne(quil_prog, lambda p: expectation_fn(theta, p), fac)
+    )
+```
++++
+
+Plot the energy landscape with ZNE
+
+```{code-cell}
+plt.figure(2)
+plt.plot(thetas, results_zne, 'o-')
+plt.xlabel(r'$\theta$', fontsize=18)
+_ = plt.ylabel(r'$\langle \Psi(\theta) | \frac{1 - Z}{2} | \Psi(\theta) \rangle$', fontsize=18)    
+plt.title('Mitigated Energy Landscape')
+```
++++
+
+Run optimization with ZNE
+
+```{code-cell}
+def mitigated_expectation(thetas, executable: Program, factory) -> float:
+    mitigated_exp =  zne.execute_with_zne(quil_prog, lambda p: expectation(thetas, p), factory)
+    return mitigated_exp
+
+quil_prog = qc.compiler.quil_to_native_quil(program, protoquil=True)
+res_zne = optimize.minimize(mitigated_expectation, init_angle, args=(quil_prog, fac), method='COBYLA')
+print(res_zne)
+```
++++
+
+## References
+[1] [Parametric compilation tutorial in pyQuil.]
+https://pyquil-docs.rigetti.com/en/v2.28.1/migration3-declare.html
+
+
++++
+
+display information about Mitiq, packages, and Python version/platform
+```{code-cell}
+mitiq.about()
+```
+
++++
+
+```{code-cell}
+plt.show()
+```
++++
```

Comment:
actually it needed to be moved back to the blocks that generate the plots. I moved it to the end because the program would wait for me to close the figure window before continuing, but I forgot to move it back. It's fixed now.